test/WebSites/InProcessWebSite/Startup.cs

```diff
@@ -381,6 +385,23 @@ private void ReadAndWriteEcho(IApplicationBuilder app)
             });
         }
 
+        private void ReadAndWriteEchoLines(IApplicationBuilder app)
+        {
+            app.Run(async context => {
```

Comment:
few other potential tests (some may not be applicable)
- Write to response body, then upgrade the connection
- Flush after upgrade
- Reading and Writing concurrently with websockets (two separate tasks reading and writing).

Reply:
I'd like to get this PR in, there are always more things we can test.